File: CSharpCore/HamaraBasketTests/QualityCheckProviderTests.cs
using HamaraBasketQualityAssurance;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace HamaraBasketTests
{
    [TestClass]
    public class QualityCheckProviderTests
    {
        [TestMethod]
        public void ShouldGetTheGeneralQualityChecker()
        {
            var updater = QualityCheckProvider.GetChecker(ItemType.Expiring);
            Assert.IsInstanceOfType(updater,typeof(QualityChecker));
        }

        [TestMethod]
        public void ShouldGetTheLegendaryQualityChecker()
        {
            var updater = QualityCheckProvider.GetChecker(ItemType.Legendary);
            Assert.IsInstanceOfType(updater, typeof(LegengaryItemQualityChecker));
        }

        [TestMethod]
        public void ShouldGetTheNonExpiryQualityChecker()
        {
            var updater = QualityCheckProvider.GetChecker(ItemType.NonExpiry);
            Assert.IsInstanceOfType(updater, typeof(NonExpiryItemQualityChecker));
        }
    }
}

File: Java/src/test/java/com/hamaraBasket/QualityCheckProviderTests.java
package com.hamaraBasket;

import static org.hamcrest.CoreMatchers.instanceOf;
import static org.junit.Assert.assertThat;

import org.junit.Test;
public class QualityCheckProviderTests
    {
       @Test
        public void shouldGetTheGeneralQualityChecker()
        {
            QualityChecker updater = QualityCheckProvider.GetChecker(ItemType.Expiring);
            assertThat(updater, instanceOf(QualityChecker.class));
        }

        @Test
        public void shouldGetTheLegendaryQualityChecker()
        {
            QualityChecker updater = QualityCheckProvider.GetChecker(ItemType.Legendary);
            assertThat(updater, instanceOf(LegengaryItemQualityChecker.class));
        }

        @Test
        public void shouldGetTheNonExpiryQualityChecker()
        {
            QualityChecker updater = QualityCheckProvider.GetChecker(ItemType.NonExpiry);
            assertThat(updater, instanceOf(NonExpiryItemQualityChecker.class));
        }
    }